apps/internal/oauth/resolvers.go

```diff
@@ -120,8 +120,10 @@ func (m *authorityEndpoint) addCachedEndpoints(authorityInfo authority.Info, use
 }
 
 func (m *authorityEndpoint) openIDConfigurationEndpoint(ctx context.Context, authorityInfo authority.Info, userPrincipalName string) (string, error) {
-	if authorityInfo.Tenant == "adfs" {
+	if authorityInfo.AuthorityType == authority.ADFS {
 		return fmt.Sprintf("https://%s/adfs/.well-known/openid-configuration", authorityInfo.Host), nil
+	} else if authorityInfo.AuthorityType == authority.DSTS {
```

Comment:
one more remark: I probably would not have to add separate dSTS flow here, if the URL was correctly parsed && I would have the option to disable authority validation, but the method that is preparing `authorityInfo` struct is called from `base.New` as

```go
// By default, validateAuthority is set to true and instanceDiscoveryDisabled is set to false
authInfo, err := authority.NewInfoFromAuthorityURI(authorityURI, true, false)
```

so there's no way for the caller to disable the validation / instance discovery

without going too crazy on the refactors here, I think there should be _some_ way for the caller to alter this

from the top of my head there is `confidential.clientOptions` which already have `Option`-altering methods, there already is `disableInstanceDiscovery` property (not propagated further), we could add `disableAuthorityValidation` + `Option` method and propagate those?

sadly, I don't know why the default in `base` pkg is true/false, the comment does not elaborate on it